src/main/webapp/mapsScript.js

```diff
@@ -23,6 +23,9 @@ var map;
 var directionsService;
 var directionsRenderer;
 
+// Map marker label constant
+var labels = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
```

Comment:
This labeling is implicitly dependent on the fact that maps routing labels it's points as A->Z, instead of, eg: 1->N or some other labeling scheme. That part of the maps api is probably not going to change anytime soon, and I don't think you need to change your implementation, but I think it's still worth noting that this is potentially fragile.

Reply:
Yes this is true! Ideally I would have liked to just change the origin marker but it seems like the API is structured such that it either automatically adds all or none of the markers and the directionsRenderer is not very transparent, so this is the best workaround I could find.